### PR TITLE
Build nickel on different rust channels

### PR DESCRIPTION
**Pull request by garbas**
- new input (`nixpkgs-mozilla`) added because of its rust overlay
- `scripts/update_channel.sh` added 
- added a daily cron job via GitHub Actions to update channels

### CODE REVIEW

**Comment by garbas** [comment]
Oh I just realized we're using circleci, I can switch to it if this is something that you'd like.

**Comment by mboes** [comment]
Standardizing on GitHub Actions (in a separate PR) sounds like a better direction to me.

**Review by yannham — COMMENTED** [review]
Great, thanks @garbas ! I just made a suggestion to make `rust-analyzer` and co work inside the shell, which I let you double-check.

File: flake.nix [hunk 1]
```diff
@@ -1,58 +1,82 @@
...
+          pkgs = import nixpkgs {
+            inherit system;
+            overlays = [ nixpkgs-mozilla.overlays.rust ];
+          };
 
-        buildPhase = "cargo build --release --frozen --offline";
+          rust =
+            (pkgs.rustChannelOf CHANNELS."${channel}").rust.override({
+              # targets = [];
+              # extentions = [];
```
[the remark this answers is not in the document]
Comment (garbas) [reply flake.nix:37]: Available extensions are depending on different channel/version. I will try to build against all 3 channels to see if your suggestions work.

**Review by yannham — APPROVED** [review]

**Pull request merged**